Assets/World/Units/Ships/Pirates/Pirate/Pirate.gd

```diff
@@ -0,0 +1,217 @@
+tool
+extends Ship
+class_name Pirate
+
+const PIRATE_IDLE_ANIM = preload("res://Assets/World/Units/Ships/Pirates/Pirate/Sprites/Pirate_idle.png")
+
+const PIRATE_BLACK_MOVE_0 = preload("res://Assets/World/Units/Ships/Pirates/Pirate/Sprites/PirateBlack_move_0.png")
+const PIRATE_BLACK_MOVE_45 = preload("res://Assets/World/Units/Ships/Pirates/Pirate/Sprites/PirateBlack_move_45.png")
+const PIRATE_BLACK_MOVE_90 = preload("res://Assets/World/Units/Ships/Pirates/Pirate/Sprites/PirateBlack_move_90.png")
+const PIRATE_BLACK_MOVE_135 = preload("res://Assets/World/Units/Ships/Pirates/Pirate/Sprites/PirateBlack_move_135.png")
+const PIRATE_BLACK_MOVE_180 = preload("res://Assets/World/Units/Ships/Pirates/Pirate/Sprites/PirateBlack_move_180.png")
+const PIRATE_BLACK_MOVE_225 = preload("res://Assets/World/Units/Ships/Pirates/Pirate/Sprites/PirateBlack_move_225.png")
+const PIRATE_BLACK_MOVE_270 = preload("res://Assets/World/Units/Ships/Pirates/Pirate/Sprites/PirateBlack_move_270.png")
+const PIRATE_BLACK_MOVE_315 = preload("res://Assets/World/Units/Ships/Pirates/Pirate/Sprites/PirateBlack_move_315.png")
+
+const PIRATE_MOVE_ANIM = [
+	PIRATE_BLACK_MOVE_0,
+	PIRATE_BLACK_MOVE_45,
+	PIRATE_BLACK_MOVE_90,
+	PIRATE_BLACK_MOVE_135,
+	PIRATE_BLACK_MOVE_180,
+	PIRATE_BLACK_MOVE_225,
+	PIRATE_BLACK_MOVE_270,
+	PIRATE_BLACK_MOVE_315,
+]
+
+export(bool) var debug_is_moving = is_moving
+
+var current_anim_position: float = 0.0
+var last_anim = "fade_out"
+
+onready var _reflection: Sprite3D = $Reflection as Sprite3D
+onready var water_overlay: Spatial = $WaterOverlay as Spatial
+onready var water_overlay1: Sprite3D = $WaterOverlay/WaterOverlay1 as Sprite3D
+onready var water_overlay2: Sprite3D = $WaterOverlay/WaterOverlay2 as Sprite3D
+onready var animation_player: AnimationPlayer = $AnimationPlayer as AnimationPlayer
+
+# Static (for now)
+onready var patrol_route = [
+	Vector3(-30, 0, 30),
+	Vector3(39, 0, 40),
+	Vector3(39, 0, -38),
+	Vector3(-35, 0, -38)
+]
+var next_patrol_index = -1
+
+func _ready() -> void:
+	if not Engine.is_editor_hint():
+		debug_is_moving = false
+
+func _process(_delta: float):
+	patrol() # DEBUG
+	
+	# TODO: Water reflection (in animate_movement())
+	# 		and WaterOverlay effect
+	#animate_water_overlay()
+
+func patrol():
+	if Engine.is_editor_hint():
+		return
+
+#	prints("pirate:",
+#		int(round(global_transform.origin.x)),
+#		int(round(global_transform.origin.y)),
+#		int(round(global_transform.origin.z))
+#	)
+#
+#	prints("patrol_route", 
+#		int(round(patrol_route[next_patrol_index].x)),
+#		int(round(patrol_route[next_patrol_index].y)),
+#		int(round(patrol_route[next_patrol_index].z))
+#	)
+	if path.size() == 0:
+		next_patrol_index += 1
+		if next_patrol_index >= patrol_route.size():
+			next_patrol_index = 0
+		move_to(patrol_route[next_patrol_index])
+
+func animate_movement():
```

Comment:
The [GDScript style guide](https://docs.godotengine.org/en/latest/getting_started/scripting/gdscript/gdscript_styleguide.html) suggests having 2 empty lines between functions. We don't have to follow everything on the guide, but it would be good if we roughly stick to it unless we have a good reason not to.

Reply:
It's the one thing I deliberately violate in the GDScript style guide (and PEP-8). I personally can't share the visibility problem that made this rule happen in the first place.
Instead it feels kind of unaesthetic having two identical lines next to each other that do essentially nothing.

I'll stick to the current formatting (and because this is right now the consensus in the codebase) unless @LinuxDonald explicitly states otherwise for his project.

In case this is too early to decide at this point though, we can still regex search & replace to get one formatting or another in place at a later stage if the respective other style turns out to be better suited for this project.